RB & FF:  Update with changes from 202202

spbrogan opened the pull request:
## Description

Cherry-pick a few changes that were in 202202 but not in 202208


## How This Was Tested

none

## Integration Instructions

none

apop5 committed:
User/apop5/updatedocument (#62)

* Updated for additional integration instructions
* Updated integration guide to add certificate information
* Updated formatting for linter

(cherry picked from commit 7f0e0750fb750a8794403e1dc35ba778dbe7e18b)

antklein committed:
Merged PR 6136: FrameBufferMemDrawLibDxe: set mGraphicsOutput to STATIC definition

## Description

* Update FrameBufferMemDrawLibDxe.c version of mGraphicsOutput to a STATIC definition
-- This definition clashes with the definition in Common\SURFACE\MsSurfaceCorePkg\Universal\TimeoutSpinner\ScreenGraphics.c

- [ ] Breaking change? No
  - Will this change break pre-existing builds or functionality without action being taken?

## How This Was Tested

Boot tested on Arcata and Lucien

## Integration Instructions

N/A

(cherry picked from commit c47b08c70c07c388cf3c8f53516b22ae414af648)

spbrogan commented on MfciPkg/Docs/Mfci_Integration_Guide.md:
````diff
@@ -33,15 +33,57 @@ Add the following to your EDK2 Flash Descriptor File (FDF)
 !include MfciPkg/MfciPkg.dsc.inc
 ```
 
+Mfci is still considered pre-production, so OPT_INTO_MFCI_PRE_PRODUCTION will need to be set at the top of the Dsc file
````
Is this a merge conflict?  I thought we removed this text from 202202?  @kuqin12  @apop5

apop5 replied:
Yes, I errantly added them back. 

I will put together a PR to drop them again.

kuqin12 replied:
This should not be here.

jeffglaum committed:
SUIT List Box element should support cell selection via keyboard even when cell type isn't checkbox. (#76)

## Description

Keyboard navigation of the SUIT List Box element should support cell
selection even when the cell type isn't checkbox (as it is for the
FrontPage boot menu options, for example). This change separates the
cell selection from the possible toggling action of a checkbox.

- [ ] Breaking change?
- Will this change break pre-existing builds or functionality without
action being taken? This is not a breaking change.

## How This Was Tested

Tested using FrontPage and a keyboard (both the main frame list box and
the boot selection list box). Also tested with a custom application that
uses a list box implementation to show Wi-Fi access points (cBMR
application which will be pushed to MU_OEM_SAMPLE shortly). Now list
selections can be made with the space bar and only if the type is
checkbox will the toggle event be issued.

## Integration Instructions

NA.

(cherry picked from commit f1aeb469511e6a72847c79ae2dc271390613d162)

makubacki reviewed (approved):
Did not compare contents, just commit difference between 202202 and 202208 branch.

jeffglaum committed:
Bug: SUIT bitmap draw was happening outside the SWM context. (#75)

The bitmap element was using the GOP protocol to render. This could be
perceived as a collision with the active window, thus forcing either a
mistaken back-buffer swap or at best an unnecessary screen refresh.

## Description

Changed the SUIT bitmap element RenderBitmap routine to use the SWM
version of the BLT function. This is what all the other SUIT elements do
by design.

- [ ] Breaking change?
- Will this change break pre-existing builds or functionality without
action being taken? This is not a breaking change and should be
invisible to SUIT consumers.

## How This Was Tested

Tested with a custom UEFI application that renders a bitmap on the
screen (cBMR client application which will be pushed to MU_OEM_SAMPLE
soon). Note that the FrontPage implementation no longer appear to show
compliance labels via the VFR form so it was not tested with FrontPage.

## Integration Instructions

NA

(cherry picked from commit 974bb2748524e24dc8f0ead06d599960ce64c0df)

apop5 committed:
Drop pre-production section from MFCI documentation (#79)

(cherry picked from commit ad83f535b951a295f6bb391e6aa54a05078feb15)
